packages/core/src/toPromise.ts

```diff
@@ -0,0 +1,47 @@
+import { AnyActorRef, OutputFrom } from './types.ts';
+
+/**
+ * Returns a promise that resolves to the `output` of the actor when it is done.
+ *
+ * @example
+ * ```ts
+ * const machine = createMachine({
+ *   // ...
+ *   output: {
+ *     count: 42
+ *   }
+ * });
+ *
+ * const actor = createActor(machine);
+ *
+ * actor.start();
+ *
+ * const output = await toPromise(actor);
+ *
+ * console.log(output);
+ * // logs { count: 42 }
+ * ```
+ */
+export function toPromise<T extends AnyActorRef>(
+  actor: T
+): Promise<OutputFrom<T>> {
+  // TODO: this is typed as `any`
```

Comment:
We can't do much about it as long as we use `T extends AnyActorRef` pattern. You unwrap here what you provided for the `TSnapshot` within `ActorRef` but that's just `any` (since you are using `AnyActorRef`)